text/3123-rustdoc-scrape-examples.md

```diff
@@ -0,0 +1,137 @@
+- Feature Name: `rustdoc_scrape_examples`
+- Start Date: 2021-05-09
+- RFC PR: [rust-lang/rfcs#3123](https://github.com/rust-lang/rfcs/pull/3123)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+This RFC proposes an extension to Rustdoc that automatically scrapes code examples from the project's `examples/` directory. 
+
+Check out a live demo here: https://willcrichton.net/example-analyzer/warp/trait.Filter.html#method.and
+
+# Motivation
+[motivation]: #motivation
+
+Code examples are an important tool for programmers to understand how a library works. Examples are **concrete** and **contextual**: they reference actual values rather than abstract parameters, and they show how to use a function in the context of the code around it. 
+
+As a parable of the value of examples, I recently did a small user study where I observed two Rust programmers learning to use the [warp](https://github.com/seanmonstar/warp) library for a basic task. Warp is designed around a generic [`Filter`](https://docs.rs/warp/0.3.1/warp/trait.Filter.html) abstraction. Both participants found the documentation for `Filter` methods to be both imposing and too generic to be useful. For instance, [`Filter::and`](https://docs.rs/warp/0.3.1/warp/trait.Filter.html#method.and):
+
+<kbd>
+<img width="600" alt="Screen Shot 2021-05-09 at 7 49 35 PM" src="https://user-images.githubusercontent.com/663326/117592915-fe07b880-b0ff-11eb-97e9-43197fbcb2a7.png">
+</kbd>
+<br /><br />
```

Comment:
Eek, the markup! My eyes!

- Please put real descriptive alt text, even if brief. (I will admit that this is one of the rare occasions that “Screen Shot *‹time›*” actually *isn’t* completely bad alt text, but it’s still nasty.)

- Please don’t abuse `<kbd>` like this to get a border, it makes me sad. Plus it doesn’t work when published in https://rust-lang.github.io/rfcs/ (but please don’t patch that to similarly style it, or I might cry!)

- The images are part of the RFC, so I’d say they should be included in the repository, not externally-hosted; but apparently there’s no solid way of doing that at present, so I’ve just filed #3172.

- Also I don’t like the `<br /><br />`; the manual line breaks don’t feel called for. (Incidentally also also, the trailing slash is an XHTML anachronism, misleading in HTML because it doesn’t close elements, but is just completely ignored by the parser. You’re better to write `<br>`.)

Reply:
I've simplified the image markup and added alt text. I left the image links as-is until your issue is resolved.